### PR TITLE
Update c/image for sysregistriesv2 changes and automatic docker:// insecure configuration

### PR DESCRIPTION
**Pull request by mtrmac**
This updates c/image to include containers/image#468 , buildah to include https://github.com/containers/buildah/pull/1214, and updates the callers.

In addition to the straightforward API change, it also rips out the existing code manually handling the list of insecure registries; this should now be handled by the `docker://` transport.

See the individual commit messages for details.

The way `types.SystemContext.SystemRegistriesConfPath` is set continues to be fairly irregular, only at the places that immediately need it; this really feels like something that should be set in one (and no more than one) shared helper, and `types.SystemContext` values should be passed along either the call stack or the data structures (`Runtime`?); this PR only makes a localized transition, admits to it via added FIXMEs, and does not have the ambition to rework the `SystemContext` use at all.

Warning: I haven’t tested this manually one bit.

This is a prerequisite to merging the blob-info-caching branch.

### CODE REVIEW

File: pkg/registries/registries.go [hunk 1]
```diff
@@ -13,21 +13,28 @@ import (
...
+// SystemRegistriesConfPath returns an appropriate value for types.SystemContext.SystemRegistriesConfPath
+// (possibly "", which is not an error), taking into account rootless mode and environment variable overrides.
+//
+// FIXME: This should be centralized in a global SystemContext initializer inherited throughout the code,
+// not haphazardly called throughout the way it is being called now.
+func SystemRegistriesConfPath() string {
 	if rootless.IsRootless() {
 		if _, err := os.Stat(userRegistriesFile); err == nil {
-			registryConfigPath = userRegistriesFile
+			return userRegistriesFile
```
Comment (giuseppe) [inline pkg/registries/registries.go]: previously the `REGISTRIES_CONFIG_PATH` environment variable was honored also for the rootless case.

It is probably enough to change the order of the checks

File: libpod/image/image.go [hunk 2]
```diff
@@ -546,7 +546,7 @@ func (i *Image) PushImageToReference(ctx context.Context, dest types.ImageRefere
 		}
 		registry := reference.Domain(imgRef)
 
-		if util.StringInSlice(registry, insecureRegistries) && !forceSecure {
+		if util.StringInSlice(registry, insecureRegistries) && dockerRegistryOptions.DockerInsecureSkipTLSVerify != types.OptionalBoolFalse {
```
[the remark this answers is not in the document]
Comment (mtrmac) [reply libpod/image/image.go]: (Or, maybe, at a higher level, something roughly like `TLSVerificationDisabled`, to also clarify he double negatives in `SkipVerify`…. And, independently, in `podman`, a shared helper for the `!c.BoolT("tls-verify")` to make sure there can’t be simple typos… — or, ultimately, a CLI rework that does not rely on the `"tis-verify"` string.

Luckily, this particular line was completely removed in one of the next commits, in favor of `sysregistriesv2` doing the right thing, and no check of an optional boolean is the best kind of check of an optional boolean :) )

File: docs/podman-login.1.md [hunk 3]
```diff
@@ -43,7 +43,9 @@ Default certificates directory is _/etc/containers/certs.d_.
 
 **--tls-verify**
 
-Require HTTPS and verify certificates when contacting registries (default: true)
+Require HTTPS and verify certificates when contacting registries (default: true). If explicitly set to true,
+then tls verification will be used, If set to false then tls verification will not be used. If not specified
```
[the remark this answers is not in the document]
Comment (mtrmac) [reply docs/podman-login.1.md]: Thanks, updated all 5 man pages with `--tls-verify` similarly.

**Comment by rhatdan** [comment]
Nice work @mtrmac and @vrothberg

**Comment by rhatdan** [comment]
@mtrmac Needs a rebase.

**Comment by mtrmac** [comment]
Rebased, tests are passing.

**Comment by rhatdan** [comment]
@mtrmac Sadly we got a conflict.

**Comment by mtrmac** [comment]
Rebased again; non-Linux builds are broken, so absolutely untested.

**Comment by mtrmac** [comment]
Tests are passing. PTAL

**Pull request merged**